account/signer.go

```diff
@@ -0,0 +1,54 @@
+package account
+
+import (
+	"context"
+	"math/big"
+
+	"github.com/NethermindEth/starknet.go/curve"
+	"github.com/NethermindEth/starknet.go/utils"
+)
+
+type Signer struct {
+	keystore  *MemKeystore
+	publicKey string
+}
+
+func NewSigner(privateKey *big.Int) (*Signer, error) {
+	pubKey, err := getPublicKey(privateKey)
+	if err != nil {
+		return nil, err
+	}
+
+	keyStore := SetNewMemKeystore(pubKey, privateKey)
+
+	return &Signer{
+		keystore:  keyStore,
+		publicKey: pubKey,
+	}, nil
+}
+
+func (s *Signer) PublicKey() string {
+	return s.publicKey
+}
+
+func (s *Signer) MemKeyStore() *MemKeystore {
+	return s.keystore
+}
+
+func (s *Signer) Put(priv *big.Int) {
```

Comment:
Same as above, let's request privateKey as a string. Also, this method should allow pointing an existing MemKeyStore to this signer, that way the dev could create only one keystore map and have multiple signers using it. It's like creating a new signer but also passing an existing MemKeyStore pointer to it

Reply:
Hey @thiagodeev 
didn't get what u said about existing MemKeyStore...
I pushed the changes. please look whether you talking about it or something else??

Reply:
If you take a look at the MemKeyStore struct you'll notice that it is a map[string]*big.Int, so when we instantiate a new Signer here we're just creating a new map and setting our pub and priv key as key/values in that map.
So, if the user already has a MemKeyStore instantiated storing other keys, why not use this existing map to store new keys instead of creating a new one?
Makes sense to you now?

Reply:
done